### ISSUE TITLE
Structural subtyping use case incorrectly type-checks

### ISSUE THREAD

**Issue by sanjioh** [microsoft/pyright#929]
**Describe the bug**
Inheritance from mixin class incorrectly type-checks. 

**To Reproduce**
```python
from typing import Protocol


class HasItemProtocol(Protocol):
    @property
    def item(self) -> int: ...


class Mixin:
    def printitem(self: HasItemProtocol):
        print(self.item)


class A(Mixin):
    item = 1


class B(Mixin):
    pass


A().printitem()
B().printitem()
```

**Expected behavior**
pyright should report an error related to structural subtyping, since class B doesn't have the attribute specified by the `Protocol`.

**VS Code extension or command-line**
pyright 1.1.61, invoked from CLI.

Thanks for your support, please let me know if I can provide further information.

**Comment by erictraut** [comment]
Thanks for the bug report. 

You are correct. Currently, pyright doesn't validate that a custom-annotated "self" type is honored when a method is "bound" to an instance. In this case, this binding is done by the expression `A().printitem`.

**Comment by sanjioh** [comment]
Ok, thanks to you for the rapid feedback! 👍🏻

**Comment by erictraut** [comment]
I've implemented the fix, and it will be in the next version of Pyright.

**Comment by sanjioh** [comment]
Awesome, thank you.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.62, which I just published. It will also be included in the next version of Pylance.
